### DISCUSSION TITLE
Handling comments in Vim files.

### DISCUSSION THREAD

**Discussion by theTyster** [LunarWatcher/auto-pairs#87] [Q&A] [answered]
Vim uses the `"` character to start a comment line.

I find that removing comments can be frustrating because of this. For example:

```vim
0: " This is a silly remap. | <-- my cursor.
1: nnoremap <CR> :echo "pew"<CR>
```
Let's say that I am on line `0` somewhere. I hit `O` to start a new line above the comment. So now my file looks like this:

```vim
0: " |
1: " This is a silly remap.
2: nnoremap <CR> :echo "pew"<CR>
```
This is expected. As in other languages, when starting a newline from a commented line that new line begins with a comment as well.

### But now things get weird.

Let's say that I don't actually want this new line to be commented out so I hit `<BS>` to begin to remove it. What I end up with is this:

```vim
0: "|" This is a silly remap.
1: nnoremap <CR> :echo "pew"<CR>
```
#### Whoops! That's weird! I guess I'll just delete the comment character in front of my cursor and fix this:

```vim
0: |This is a silly remap.
1: nnoremap <CR> :echo "pew"<CR>
```
Oh right, I'm using autopairs and so it deleted them both. Ok I'll rewrite the comment:

```vim
0: "|This is a silly remap.
1: nnoremap <CR> :echo "pew"<CR>
```
Now, we are back where we started. So we rinse and repeat one more time before I realize that I have to do something special because of how `"` characters are handled by autopairs.

My workarounds usually look something like:

```vim
normal O<esc>S
```
or if there's lines above it *that aren't commented out*:
```vim
normal kko
```

# End Rant.
I assume there is some kind of simple solution to this. Because otherwise I imagine users of autopairs like LunarWatcher would go crazy as they wrote their plugins.

So, I'd like to see what some people are doing to work around or resolve this kind of behavior. All ideas are welcome!

**Comment by LunarWatcher** [comment]
I can't reproduce. What config are you using?

**Reply by theTyster** [reply, marked as answer]
@LunarWatcher 
Oh, yay. I have once again publicly demonstrated my intelligence.

It appears that I left a buffer open with [jiangmiao/auto-pairs](https://github.com/jiangmiao/auto-pairs)'s plugin after adding your plugin and removing his.

We're all good.

Sorry for bothering you about this. Thanks for taking time to respond.
